addons/sourcemod/scripting/vsh/bosses/boss_pyromancer_scalded.sp

```diff
@@ -0,0 +1,239 @@
+	"vo/pyro_jeers02.mp3"
+};
+
+static char g_strPrecacheCosmetics[][] =
+{
+	"models/player/items/pyro/pyro_pyromancers_mask.mdl",
+	"models/player/items/pyro/hwn_pyro_misc1.mdl",
+	"models/player/items/pyro/sore_eyes.mdl",
+	"models/workshop/player/items/pyro/hw2013_dragonbutt/hw2013_dragonbutt.mdl"
+};
+
+static int g_iCosmetics[] =
+{
+	316,
+	550,
+	387,
+	30225
+};
+
+static int g_iPrecacheCosmetics[4];
+
+static float g_flFlamethrowerRemoveTime[TF_MAXPLAYERS+1];
+
+static CRageAddCond addCond;
+
+methodmap CScaldedPyromancer < SaxtonHaleBase
+{
+	public CScaldedPyromancer(CScaldedPyromancer boss)
+	{
+		boss.CallFunction("CreateAbility", "CBraveJump");
+		addCond = boss.CallFunction("CreateAbility", "CRageAddCond");
+		addCond.flRageCondDuration = RAGE_DURATION;
+		
+		//boostJump.flMaxHeigth /= 1.75;
+		//boostJump.flMaxDistance = 0.7;
+		
+		boss.iBaseHealth = 500;
+		boss.iHealthPerPlayer = 750;
+		boss.nClass = TFClass_Pyro;
+		boss.iMaxRageDamage = 1700;
+		g_flFlamethrowerRemoveTime[boss.iClient] = 0.0;
+	}
+	
+	public void GetBossName(char[] sName, int length)
+	{
+		strcopy(sName, length, "Scalded Pyromancer");
+	}
+	
+	public void GetBossInfo(char[] sInfo, int length)
+	{
+		StrCat(sInfo, length, "\nDuo Boss: The Pyromancers");
+		StrCat(sInfo, length, "\nMelee deals 80 damage.");
+		StrCat(sInfo, length, "\nHealth: Low");
+		StrCat(sInfo, length, "\n ");
+		StrCat(sInfo, length, "\nAbilities");
+		StrCat(sInfo, length, "\n- Boost Jump");
+		StrCat(sInfo, length, "\n ");
+		StrCat(sInfo, length, "\nRage");
+		StrCat(sInfo, length, "\n- Grants a degreaser for 8 seconds");
+		StrCat(sInfo, length, "\n- 200%% Rage: Grants a buffed backburner with quick-switch for 8 seconds");
+	}
+	
+	public void OnSpawn()
+	{
+		const int TF_WEAPON_AXTINGUISHER = 38;
+		int iWeapon = this.CallFunction("CreateWeapon", TF_WEAPON_AXTINGUISHER, "tf_weapon_fireaxe", 100, TFQual_Collectors, "2 ; 1.82 ; 20 ; 1.0 ; 252 ; 0.5");
+		if (iWeapon > MaxClients)
+			SetEntPropEnt(this.iClient, Prop_Send, "m_hActiveWeapon", iWeapon);
+		
+		/*
+		Axtinguisher attributes:
+		
+		2: damage bonus
+		20: crit vs burning players
+		252: reduction in push force taken from damage
+		*/
+		
+		for (int i = 0; i < sizeof(g_iCosmetics); i++)
+		{
+			int iWearable = this.CallFunction("CreateWeapon", g_iCosmetics[i], "tf_wearable", 1, TFQual_Collectors, "");
+			if (iWearable > MaxClients)
+			{
+				SetEntProp(iWearable, Prop_Send, "m_nModelIndexOverrides", g_iPrecacheCosmetics[i]);
+				
+				if (i == 0) //Pyromancer's Mask
+				{
+					SetEntProp(iWearable, Prop_Send, "m_nSkin", 2);
+					SetEntityRenderColor(iWearable, 0, 0, 255, 200);
+				}
+				
+				if (i == 3) //Cauterizer's Caudal Appendage
+				{
+					SetEntityRenderColor(iWearable, 0, 0, 255, 255);
+				}
+			}
+		}
+		
+		PrintHintText(this.iClient, "HINT: Stay near the other Pyromancer so you can crit the ignited players!");
+	}
+	
+	public void OnRage()
+	{
+		addCond.AddCond(TFCond_Buffed);
```

Comment:
You only need to call this in constructor, then `CRageAddCond` will do the work for you. Otherwise your stacking `TFCond_Buffed` array on every rage